### PR TITLE
Added a benchmark for game logic

### PR DESCRIPTION
**Pull request by frutiemax**
Google benchmark parameters seem to work. Still need to figure out what to do with CPU iterations. The command to be called is openrct2 benchlogic xxx.sv6

### CODE REVIEW

File: src/openrct2/cmdline/BenchLogic.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#include "../GameState.h"
+
+
+#include <cstdlib>
+#include <chrono>
+#include <memory>
+
+using namespace OpenRCT2;
+
+static void BM_gamelogic(benchmark::State& state, IContext* context)
```
Comment (janisozaur) [inline src/openrct2/cmdline/BenchLogic.cpp]: Why can't it use the standard form
```
for (auto _ : state)
{
    context->GetGameState()->UpdateLogic();
}
```
?

**Comment by janisozaur** [comment]
Other than the comment I posted above:
1. Make sure you formatted your code
2. Make sure your commit message conforms to https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages

The xcode failure needs to be handled by someone with access to xcode. Ping @AaronVanGeffen @Gymnasiast

File: src/openrct2/cmdline/BenchLogic.cpp (hunk 1 above)
Comment (frutiemax) [reply src/openrct2/cmdline/BenchLogic.cpp]: Actually, you found the problem with my code, you actually need this. Why?

**Comment by janisozaur** [comment]
That's not what I meant. I think in current form it will cause problems because you enabled manual timing and provide your own measurements, while all this should be handled by `state`. Why is it necessary to do your own timing instead of using the benchmark's one?

**Comment by frutiemax** [comment]
It is not necessary to do manual timing. But it works nonetheless and it is one of the examples in the documentation of the Google Benchmark library.

**Comment by janisozaur** [comment]
Since it is not necessary, I would prefer to use the simpler invocation

**Comment by AaronVanGeffen** [comment]
@frutiemax I've rebased this PR and pushed a fix for the Xcode project. Do you intend to continue to work on this?

File: src/openrct2/cmdline/BenchLogic.cpp (hunk 1 above)
Comment (chfast) [reply src/openrct2/cmdline/BenchLogic.cpp]: The loop is actually measuring the time, you don't need your timer. I'm just not sure it is safe to call `context->GetGameState()->UpdateLogic()` repeatedly in a loop.

File: src/openrct2/cmdline/BenchLogic.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
```
Comment (chfast) [inline src/openrct2/cmdline/BenchLogic.cpp]: You also don't need to call the function with `BM_` prefix.

**Comment by AaronVanGeffen** [comment]
This is essentially doing the same thing as #10700. We need to decide which to merge.

**Pull request closed without merging**